tests/jballerina-debugger-integration-test/src/test/resources/project-based-tests/variable-tests/main.bal

```diff
@@ -334,3 +342,15 @@ function printDetails(string name, int age = 18, string... modules) returns stri
 function addition(int a, int b) returns int {
     return a + b;
 }
+
+function basicClosure() returns (function (int) returns int) {
+    int a = 3;
+    var foo = function (int b) returns int {
+        int c = 34;
+        if (b == 3) {
+            c = c + b + a + port;
+        }
+        return c + a;
+    };
+    return foo;
+}
```

Comment:
Shall we fix formatting in here? also please check all the newly added ballerina code snippets

Reply:
Fixed via https://github.com/ballerina-platform/ballerina-lang/pull/34432/commits/6bf21a90333a90f6b9c842810c54781836fc8ec2